ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RegexRouting.kt

```diff
@@ -0,0 +1,174 @@
+): Route = post(path) {
+    body(call.receive())
+}
+
+/**
+ * Builds a route to match `HEAD` requests with the specified regex [path].
+ * @see [Application.routing]
+ */
+@KtorDsl
+public fun Route.head(path: Regex, body: PipelineInterceptor<Unit, ApplicationCall>): Route {
+    return route(path, HttpMethod.Head) { handle(body) }
+}
+
+/**
+ * Builds a route to match `PUT` requests with the specified regex [path].
+ * @see [Application.routing]
+ */
+@KtorDsl
+public fun Route.put(path: Regex, body: PipelineInterceptor<Unit, ApplicationCall>): Route {
+    return route(path, HttpMethod.Put) { handle(body) }
+}
+
+/**
+ * Builds a route to match `PUT` requests with the specified regex [path] receiving a request body as content of the [R] type.
+ * @see [Application.routing]
+ */
+@KtorDsl
+@JvmName("putTypedPath")
+public inline fun <reified R : Any> Route.put(
+    path: Regex,
+    crossinline body: suspend PipelineContext<Unit, ApplicationCall>.(R) -> Unit
+): Route = put(path) {
+    body(call.receive())
+}
+
+/**
+ * Builds a route to match `PATCH` requests with the specified regex [path].
+ * @see [Application.routing]
+ */
+@KtorDsl
+public fun Route.patch(path: Regex, body: PipelineInterceptor<Unit, ApplicationCall>): Route {
+    return route(path, HttpMethod.Patch) { handle(body) }
+}
+
+/**
+ * Builds a route to match `PATCH` requests with the specified regex [path] receiving a request body as content of the [R] type.
+ * @see [Application.routing]
+ */
+@KtorDsl
+@JvmName("patchTypedPath")
+public inline fun <reified R : Any> Route.patch(
+    path: Regex,
+    crossinline body: suspend PipelineContext<Unit, ApplicationCall>.(R) -> Unit
+): Route = patch(path) {
+    body(call.receive())
+}
+
+/**
+ * Builds a route to match `DELETE` requests with the specified regex [path].
+ * @see [Application.routing]
+ */
+@KtorDsl
+public fun Route.delete(path: Regex, body: PipelineInterceptor<Unit, ApplicationCall>): Route {
+    return route(path, HttpMethod.Delete) { handle(body) }
+}
+
+/**
+ * Builds a route to match `OPTIONS` requests with the specified regex [path].
+ * @see [Application.routing]
+ */
+@KtorDsl
+public fun Route.options(path: Regex, body: PipelineInterceptor<Unit, ApplicationCall>): Route {
+    return route(path, HttpMethod.Options) { handle(body) }
+}
+
+private fun Route.createRouteFromRegexPath(regex: Regex): Route {
+    return this.createChild(PathSegmentRegexRouteSelector(regex))
+}
+
+public class PathSegmentRegexRouteSelector(private val regex: Regex) : RouteSelector() {
+
+    override fun evaluate(context: RoutingResolveContext, segmentIndex: Int): RouteSelectorEvaluation {
+        val prefix = if (regex.pattern.startsWith('/')) "/" else ""
+        val postfix = if (regex.pattern.endsWith('/')) "/" else ""
+        val pathSegments = context.segments.drop(segmentIndex).joinToString("/", prefix, postfix)
+        return regex.find(pathSegments)?.let { result ->
```

Comment:
personally I would prefer `val result = regex.find(pathSegments) ?: return RouteSelectorEvaluation.Failed` to remove nesting